web/src/features/service-worker/UpdatePrompt.tsx

```diff
@@ -0,0 +1,44 @@
+import { ONE_HOUR } from 'api/helpers';
+import Toast from 'components/Toast';
+import { useTranslation } from 'react-i18next';
+import { useRegisterSW } from 'virtual:pwa-register/react';
+
+function UpdatePrompt() {
+  const { t } = useTranslation();
+  const {
+    needRefresh: [needRefresh, setNeedRefresh],
+    updateServiceWorker,
+  } = useRegisterSW({
+    onRegisteredSW: (_swURL, registration) => {
+      registration &&
+        setInterval(
+          () => {
+            console.info(`Checking for app update...`);
+            registration.update();
+          },
+          import.meta.env.PROD ? ONE_HOUR : 10 * 1000
```

Comment:
Would it be a problem to check more often than once an hour? And what was the reason for doing the `prompt` approach vs `autoUpdate`?

Reply:
AutoUpdate is pretty ruthless and will update as soon as a new version is found and force reload the application, not the best UX but I suppose it's a "safer" option implementation wise.

Reply:
In the past we (somehow) had it fetch new assets in the background, but still prompt users to "reload" before it was used. But might be too much for now, we can always move that direction later on if we want :)

Reply:
That might require us to build a more custom service worker, something we can do but I'm not sure it's worth it TBH. At least not yet.